analyzer/src/main/kotlin/managers/CocoaPods.kt

```diff
@@ -19,25 +20,64 @@
 
 package org.ossreviewtoolkit.analyzer.managers
 
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties
+import com.fasterxml.jackson.core.JsonParser
+import com.fasterxml.jackson.databind.DeserializationContext
+import com.fasterxml.jackson.databind.JsonNode
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize
+import com.fasterxml.jackson.databind.deser.std.StdDeserializer
+import com.fasterxml.jackson.databind.node.ObjectNode
+
+import com.vdurmont.semver4j.Requirement
+
 import java.io.File
+import java.util.SortedSet
 
 import org.ossreviewtoolkit.analyzer.AbstractPackageManagerFactory
 import org.ossreviewtoolkit.analyzer.PackageManager
+import org.ossreviewtoolkit.downloader.VersionControlSystem
+import org.ossreviewtoolkit.model.Hash
+import org.ossreviewtoolkit.model.Identifier
+import org.ossreviewtoolkit.model.OrtIssue
+import org.ossreviewtoolkit.model.Package
+import org.ossreviewtoolkit.model.PackageReference
+import org.ossreviewtoolkit.model.Project
 import org.ossreviewtoolkit.model.ProjectAnalyzerResult
+import org.ossreviewtoolkit.model.RemoteArtifact
+import org.ossreviewtoolkit.model.Scope
+import org.ossreviewtoolkit.model.VcsInfo
+import org.ossreviewtoolkit.model.VcsType
 import org.ossreviewtoolkit.model.config.AnalyzerConfiguration
 import org.ossreviewtoolkit.model.config.RepositoryConfiguration
+import org.ossreviewtoolkit.model.createAndLogIssue
+import org.ossreviewtoolkit.model.readValue
+import org.ossreviewtoolkit.model.yamlMapper
+import org.ossreviewtoolkit.utils.CommandLineTool
+import org.ossreviewtoolkit.utils.stashDirectories
+import org.ossreviewtoolkit.utils.textValueOrEmpty
 
 /**
  * The [CocoaPods](https://cocoapods.org/) package manager for Objective-C.
- */
+ *
+ * As pre-condition for the analysis each respective definition file must have a sibling lockfile named 'Podfile.lock'.
+ * The dependency tree is constructed solely based on parsing that lockfile. So, the dependency tree can be constructed
+ * on any platform. Note that obtaining the dependency tree from the 'pod' command without a lock file has Xcode
+ * dependencies and is not supported by this class.
+ *
+ * The only interactions with the 'pod' command happen in order to obtain meta-data for dependencies. Therefore
+ * 'pod repo update' and 'pod spec which' get executed, which works also under Linux.
+ *
+ * Note: This class depends on https://github.com/CocoaPods/CocoaPods/pull/10609 which is not yet released, so a build
+ * of the master branch is needed.
```

Comment:
I would not be as specific here, esp when changing to the ZIP download above. Simply omitting the part after the comma would be fine with me.